script.js

```diff
@@ -0,0 +1,23 @@
+const expansesContainer = document.querySelector('.expenses-container');
+const totalDisplay = document.getElementById('total');
+const API = 'http://localhost:3000/api/list/';
+
+async function fetchAPI() {
+    const url = await fetch("http://localhost:3000/api/list");
+    const res = await url.json();
+    res.forEach(element => {
+     getAllExpenses(element)
+    });
+}
+fetchAPI();
+
+// GET all data
+const getAllExpenses = (data) => {
+    expansesContainer.innerHTML += `<div class="list">
+    <p class="shop">Shop "${data.name}"</p>
+    <p class="date">${data.updatedAt}</p>
+    <p class="amount">$ ${data.price}</p>
+    <img id="${data.id}" onClick="updateInstanceById()" class="edit" src="https://img.icons8.com/ios-glyphs/30/000000/edit--v1.png"/>
```

Comment:
don't add actions straight on html elements, it's bad practice, because you show guts of your app.